src/components/common/Layout/components/Account/components/Notifications/components/xmtp.ts

```diff
@@ -0,0 +1,142 @@
+
+export const useSaveXmtpKeys = () => {
+  const fetcherWithSign = useFetcherWithSign()
+  const { data: signer } = useWalletClient()
+
+  const { id } = useUser()
+
+  return useCallback(
+    async () =>
+      Client.getKeys(signer, { env: "production" }).then(async (key) =>
+        fetcherWithSign([
+          id ? `/v2/users/${id}/keys` : undefined,
+          {
+            body: {
+              key: Buffer.from(key).toString("binary"),
+              service: "XMTP",
+            },
+          },
+        ])
+      ),
+    [id, signer]
+  )
+}
+
+export const useDeleteXmtpKeys = () => {
+  const fetcherWithSign = useFetcherWithSign()
+  const { id } = useUser()
+
+  return async () =>
+    fetcherWithSign([
+      id ? `/v2/users/${id}/keys/1` : undefined,
+      { method: "DELETE" },
+    ])
+}
+
+export const useGetXmtpKeys = () => {
+  const fetcherWithSign = useFetcherWithSign()
+  const [data, setData] = useState(null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState(null)
+  const { id, error: userError, isLoading: isUserLoading } = useUser()
+
+  useEffect(() => {
+    if (id) {
+      setIsLoading(true)
+      fetcherWithSign([
+        id ? `/v2/users/${id}/keys` : null,
+        {
+          method: "GET",
+          body: {
+            Accept: "application/json",
+            query: { service: "XMTP" },
+          },
+        },
+      ])
+        .then((response) => {
+          setData(response)
+        })
+        .catch(setError)
+        .finally(() => {
+          setIsLoading(true)
+        })
+    }
+  }, [id])
+
+  return {
+    keys: data?.keys,
+    isLoading: isUserLoading || isLoading,
+    error: userError && error,
+  }
+}
+
+export const useXmtpAccessChecking = () => {
+  const [hasXmtpAccess, setHasAccess] = useState<boolean>(false)
+  const [isCheckingXmtpAccess, setIsCheckingAccess] = useState(false)
+  const { canMessageStatic } = useCanMessage()
+  const showErrorToast = useShowErrorToast()
+
+  const { address } = useAccount()
+
+  useEffect(() => {
+    if (address) {
+      setIsCheckingAccess(true)
+      canMessageStatic(address)
+        .then(setHasAccess)
+        .catch((e) => {
+          console.error("XmtpAccessCheckingError", e)
+          showErrorToast("Error happened during checking XMTP access")
+        })
+        .finally(() => setIsCheckingAccess(false))
+    }
+  }, [address, canMessageStatic])
+  return {
+    hasXmtpAccess,
+    isCheckingXmtpAccess,
+  }
+}
+
+export const useSubscribeXmtp = () => {
+  const [isSubscribingXmtp, setIsSubscribingXmtp] = useState<boolean>(false)
+
+  const toast = useToast()
+  const showErrorToast = useShowErrorToast()
+
+  const { data: signer, isLoading } = useWalletClient()
+
+  const subscribeXmtp = useCallback(async () => {
+    setIsSubscribingXmtp(true)
+    await Client.create(signer, {
+      persistConversations: false,
+      env: "production",
+    })
+      .then(() => {
+        toast({
```

Comment:
The copy isn't right here, this should be XMTP. And also, let's use the custom `useToast` hook with the customized toast:
<img width="607" alt="image" src="https://github.com/guildxyz/guild.xyz/assets/92519134/34c35235-c015-4087-8ad9-19797c0e8ce4">

I also noticed that the Sign button next to the XMTP won't get disabled after successfully enabling the identity.

Reply:
can I use the useSubmit hook here also?

Reply:
I used useSubmit for handle the loading state of this checking xmtp hook